### ISSUE TITLE
Linking to items by name doesn't work

### ISSUE THREAD

**Issue by ctron** [intellij-rust/intellij-rust#6885]
## Environment

* **IntelliJ Rust plugin version:** 0.3.141.3674-203
* **Rust toolchain version:** 1.49.0 (e1884a8e3 2020-12-29) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2020.3.2 (CL-203.7148.70)
* **Operating system:** Linux 5.10.16-200.fc33.x86_64
* **Macro expansion engine:** new
* **Name resolution engine:** old

## Problem description

Linking to items by name in the documentation seems to not work: https://doc.rust-lang.org/rustdoc/linking-to-items-by-name.html

## Steps to reproduce
```rust
/// The unix epoch, as [`chrono::DateTime`].
pub fn epoch() -> DateTime<Utc> {
    Utc.timestamp(0, 0)
}
```

I would expect to see `chrono::DateTime` linked. However I see:

![image](https://user-images.githubusercontent.com/202474/108844962-1b3ae980-75dd-11eb-974a-d8e2b19b980e.png)

**Comment by LoganDark** [comment]
Yeah, intra-doc linking is a relatively new feature that just released, just like const generics (some of it anyway). `intellij-rust` doesn't have support yet but hopefully it will soon :D
